easybuild/framework/easyconfig/tweak.py

```diff
@@ -1040,49 +1050,45 @@ def find_potential_version_mappings(dep, toolchain_mapping, versonsuffix_mapping
+                # determine search pattern based on toolchain, version prefix/suffix & version regex
+                if toolchain['name'] == DUMMY_TOOLCHAIN_NAME:
+                    toolchain_suffix = ''
+                else:
+                    toolchain_suffix = '-%s-%s' % (toolchain['name'], toolchain['version'])
+                full_versionsuffix = toolchain_suffix + versionsuffix + EB_FORMAT_EXTENSION
+                depver = '^' + prefix_to_version + candidate_ver + full_versionsuffix
                 cand_paths = search_easyconfigs(depver, consider_extra_paths=False, print_result=False,
                                                 case_sensitive=True)
-                # Filter out easyconfigs that have been tweaked in this instance (they sit in the tempdir in a subdir
-                # that starts with 'tweaked_*')
+
+                # Filter out easyconfigs that have been tweaked in this instance
+                # (they sit in the tempdir in a subdir that starts with 'tweaked_*')
                 tweaked_ec_stub = os.path.join(tempfile.gettempdir(), 'tweaked_')
                 cand_paths = [path for path in cand_paths if not path.startswith(tweaked_ec_stub)]
+
                 # Add what is left to the possibilities
                 for path in cand_paths:
-                    # Get the version from the path
-                    filename = os.path.basename(path)
-                    # Find the version sandwiched between our known values
-                    regex = re.compile('^%s(.+?)%s' % (prefix_to_version, full_versionsuffix))
-                    res = regex.search(filename)
-                    if res:
-                        version = res.group(1)
+                    version = fetch_parameters_from_easyconfig(read_file(path), ['version'])[0]
+                    if version:
                         if highest_version is None or LooseVersion(version) > LooseVersion(highest_version):
                             highest_version = version
                     else:
-                        raise EasyBuildError("Failed to determine version from '%s' using regex pattern '%s'", filename,
-                                             regex.pattern)
-                    potential_version_matches.append({'version': version, 'path': path, 'toolchain': toolchain})
+                        raise EasyBuildError("Failed to determine version from contents of %s", path)
+
+                    potential_version_matches.append({'path': path, 'toolchain': toolchain, 'version': version})
 
     if highest_versions_only and highest_version is not None:
         potential_version_matches = [d for d in potential_version_matches if d['version'] == highest_version]
 
-    _log.debug("Found possible dependency upgrades: %s", potential_version_matches)
+    _log.debug("Found potential version match for %s: %s", dep, potential_version_matches)
```

Comment:
Hmm, this name is not really helpful, I guess the variable should be `potential_version_mappings` and the text `potential version map`

Reply:
fixed in 531dd7c7b